### DISCUSSION TITLE
How to use `text-moderation-stable` ?

### DISCUSSION THREAD

**Discussion by KenobySky** [spring-projects/spring-ai#986] [Q&A]
Greetings!
Im attempting to use OpenAI Moderator on my project but everytime I try, I get the following message:
The model `text-moderation-stable` does not exist or you do not have access to it."

Am I doing something wrong?
Here's the following code to it:

```
    private boolean moderate(String userInput, OpenAiChatOptions defaultOptions) {

        OpenAiChatOptions moderatorOptions = OpenAiChatOptions.builder()
                .withModel("text-moderation-stable")
                .withUser(defaultOptions.getUser())
                .build();

        Prompt prompt = new Prompt(
                userInput,
                moderatorOptions);

        ChatResponse call = this.openAiChatModel.call(prompt);

        Generation result = call.getResult();
        String content = result.getOutput().getContent();

        System.out.println(content);

        return true;
    }
```

Here's the response I get:

> org.springframework.ai.retry.NonTransientAiException: 404 - {
>     "error": {
>         "message": "The model `text-moderation-stable` does not exist or you do not have access to it.",
>         "type": "invalid_request_error",
>         "param": null,
>         "code": "model_not_found"
>     }
> }

**Comment by ThomasVitale** [comment]
The `OpenAiChatModel` provides integration with the OpenAI Chat Completion API.
The `text-moderation-stable` is a model served via the OpenAI Moderation API, which is not currently supported by Spring AI.
There is some work in progress to implement that, see https://github.com/spring-projects/spring-ai/pull/333.
